newIDE/app/src/MainFrame/EditorContainers/HomePage/GetStartedSection/index.js

```diff
@@ -158,15 +167,16 @@ const GetStartedSection = ({ showUserChip, selectInAppTutorial }: Props) => {
     );
   };
 
-  const onQuestionnaireFinished = async (survey: UserSurveyType) => {
+  const onSurveyFinished = async (survey: UserSurveyType) => {
     try {
-      setStep('questionnaireFinished');
+      setStep('surveyFinished');
       // Artificial delay to build up expectations.
       recommendationsGettingDelayPromise.current = delay(5000);
       await Promise.all([
         onEditProfile({ survey }, preferences, { throwError: true }),
         recommendationsGettingDelayPromise.current,
       ]);
+      localStorage.removeItem(localStoreUserSurveyKey);
```

Comment:
All actions on localStorage are risky if the device doesn't allow them. I think we should put all of them in a helper/utils file and have them each wrapped up in a try catch, otherwise we're facing a risk the app crashes just because the survey cannot get saved / removed from storage.